### PR TITLE
Added pander method for data.table class

### PR DESCRIPTION
**Pull request by newtux**
[data.table](https://cran.r-project.org/web/packages/data.table/index.html)  drops row name attributes and uses keys instead. While this is useful, it makes it a bit annoying to use with pander. This function grabs the first key (if there is one), removes it from the table, and sets it as the row name attribute.

The `x <- x[ ,!(colnames(x) %in% key(test)[1])]` is not the most elegant, but I wasn't sure if there was a better way to do this.

### CODE REVIEW

File: R/S3.R [hunk 1]
```diff
@@ -238,6 +238,27 @@ pander.table <- function(x, caption = attr(x, 'caption'), ...) {
...
+#' @param ... optional parameters passed to raw \code{pandoc.table} function
+#' @export
+pander.data.table <- function(x, caption = attr(x, 'caption'), ...) {
+
+    if (is.null(caption) & !is.null(storage$caption)) {
+        caption <- get.caption()
+    }
+    if(haskey(x)){
+
+        row.names.dt <- test[[key(test)[1]]]
```
Comment (daroczig) [inline R/S3.R]: `test` should be `x` here, no? And also below.

File: R/S3.R [hunk 2]
```diff
@@ -238,6 +238,27 @@ pander.table <- function(x, caption = attr(x, 'caption'), ...) {
...
+        x <- x[ ,!(colnames(x) %in% key(test)[1])]
```
Comment (daroczig) [inline R/S3.R]: Maybe using `setdiff` for the column names is more elegant

**Comment by daroczig** [comment]
Sorry for the long delay with my reply! First of all, thank you very much for this PR, indeed makes a lot of sense, but I've just added 2 comment above -- can you please address those? Also, I think this feature should be optional, or at least there should be a way to disable that, so can we have an extra argument for `pander.data.table` to disable/enable this feature?

**Comment by newtux** [comment]
Whoops - good call on the `test` variable, it should be x. And setdiff would work much better! 

I'll address those two points and make it optional. Thanks for the feedback.

**Comment by daroczig** [comment]
Thank you!

**Pull request merged**